sympy/core/cache.py

```diff
@@ -148,3 +148,17 @@ def _getenv(key, default=None):
 else:
     raise RuntimeError(
         'unrecognized value for SYMPY_USE_CACHE: %s' % USE_CACHE)
+
+_cached_property_sentinel = object()
+
+
+def cached_property(func):
```

Comment:
It's probably a little tidier to put the sentinel inside the `cached_property` function.